pages/search.tsx

```diff
@@ -0,0 +1,38 @@
+import { useQueryState } from "next-usequerystate";
+import React from "react";
+import { useDebounce } from "react-use";
+
+import { BaseTemplate, ProductCollection } from "@/components";
+import { ProductFilterInput } from "@/saleor/api";
+
+const SearchPage = () => {
+  const [filter, setFilter] = useQueryState("search");
```

Comment:
```suggestion
  const [filter, setFilter] = useQueryState("search", "");
```

I'm not familiar with this package - does it support default value? If so, we could skip `|| ""` at line 28

Reply:
```suggestion
  const [searchQuery, setSearchQuery] = useQueryState("q");
```
IMO we can change the query argument to `q` - it's often used in other websites and makes the whole URL shorter.
I suggest changing filter to the `searchQuery` - it better describes contents of the variable (filter for products query can have more fields besides the search query)